functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -76,101 +81,74 @@ public String createWorkspace(String pathToJson, String token) throws Exception
-    StringBuilder sb = new StringBuilder(this.cheStarterURL);
-    String path = "/workspace/" + workspaceName;
-    sb.append(path);
-    sb.append("?");
-    sb.append("masterUrl=").append(this.host).append("&").append("namespace=sth");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Content-Type", "application/json");
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    OkHttpClient client = new OkHttpClient();
-    try {
-      Response response = client.newCall(requestBuilder.delete().build()).execute();
-      LOG.info("Workspace delete response : " + response.message());
-      if (!response.isSuccessful()) {
-        if (response.message().equals("Workspace not found")) {
-          LOG.warn(
-              "Workspace could not be deleteded because workspace is not found. Continuing tests.");
-          return true;
-        }
+  public boolean deleteWorkspace(String workspaceName, String token) throws IOException {
+    String relativePath = "/workspace/" + workspaceName;
+    HashMap<String, String> queryParameters = new HashMap<>();
+    queryParameters.put("masterUrl", this.host);
+    queryParameters.put("namespace", "sth");
+    String authorization = "Bearer " + token;
+    Response response =
+        cheStarterClient.sendRequest(
+            relativePath, HttpMethods.DELETE, null, authorization, queryParameters.entrySet());
+    if (response.isSuccessful()) {
+      if (response.message().equals("Workspace not found")) {
+        LOG.warn("\"Workspace could not be deleted because workspace was not found.");
       }
-      return response.isSuccessful();
-    } catch (IOException e) {
-      LOG.error("Workspace could not be deleted : " + e.getMessage(), e);
+      return true;
+    } else {
+      LOG.error("Workspace could not be deleted: " + response.message());
       return false;
     }
   }
 
-  public void startWorkspace(String WorkspaceID, String name, String token) throws Exception {
-    OkHttpClient client = new OkHttpClient();
-    String path = "/workspace/" + name;
-    StringBuilder sb = new StringBuilder(this.cheStarterURL);
-    sb.append(path);
-    sb.append("?");
-    sb.append("masterUrl=").append(this.host).append("&").append("namespace=sthf");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    RequestBody body = RequestBody.create(null, new byte[0]);
-    Request request = requestBuilder.patch(body).build();
-    try {
-      Response response = client.newCall(request).execute();
+  public void startWorkspace(String WorkspaceID, String workspaceName, String token)
+      throws Exception {
+    String relativePath = "/workspace/" + workspaceName;
+    HashMap<String, String> queryParameters = new HashMap<>();
+    queryParameters.put("masterUrl", this.host);
+    queryParameters.put("namespace", "sth");
```

Comment:
These three lines repeats in each method call. Is there some way how to generalize it?